src/test/java/io/debezium/connector/vitess/VitessConnectorIT.java

```diff
@@ -380,6 +386,64 @@ public void shouldUsePrevVgtidAsOffsetWhenNoVgtidInGrpcResponse() throws Excepti
         Testing.Print.enable();
     }
 
+    @Test
+    @FixFor("DBZ-5063")
+    public void shouldUseSameTransactionIdWhenMultiGrpcResponses() throws Exception {
+        Testing.Print.disable();
+        TestHelper.executeDDL("vitess_create_tables.ddl");
+        startConnector(config -> config.with(CommonConnectorConfig.PROVIDE_TRANSACTION_METADATA, true), false);
+        assertConnectorIsRunning();
+
+        Vgtid baseVgtid = TestHelper.getCurrentVgtid();
+        // Insert 10000 rows to make sure we will get multiple gRPC responses.
+        // The first response contains BEGIN and ROW events; The last response contains ROW, VGTID and COMMIT events.
+        int expectedRecordsCount = 10000;
+        // Expect 10000 inserts + 2 transaction metadata.
+        consumer = testConsumer(expectedRecordsCount + 2);
+        String rowValue = "(1, 1, 12, 12, 123, 123, 1234, 1234, 12345, 12345, 18446744073709551615, 1.5, 2.5, 12.34, true)";
+        StringBuilder insertRows = new StringBuilder().append("INSERT INTO numeric_table ("
+                + "tinyint_col,"
+                + "tinyint_unsigned_col,"
+                + "smallint_col,"
+                + "smallint_unsigned_col,"
+                + "mediumint_col,"
+                + "mediumint_unsigned_col,"
+                + "int_col,"
+                + "int_unsigned_col,"
+                + "bigint_col,"
+                + "bigint_unsigned_col,"
+                + "bigint_unsigned_overflow_col,"
+                + "float_col,"
+                + "double_col,"
+                + "decimal_col,"
+                + "boolean_col)"
+                + " VALUES " + rowValue);
+        for (int i = 1; i < expectedRecordsCount; i++) {
+            insertRows.append(", ").append(rowValue);
+        }
+
+        String insertRowsStatement = insertRows.toString();
+        try {
+            // exercise SUT
+            executeAndWait(insertRowsStatement);
+            String expectedTxId = assertRecordBegin();
+            for (int i = 1; i <= expectedRecordsCount; i++) {
+                SourceRecord record = assertRecordInserted(TestHelper.TEST_UNSHARDED_KEYSPACE + ".numeric_table", TestHelper.PK_FIELD);
+                final Struct txn = ((Struct) record.value()).getStruct("transaction");
+                String txId = txn.getString("id");
+                assertThat(txId).isNotNull();
+                assertThat(txId).isEqualTo(expectedTxId);
+                Vgtid actualVgtid = Vgtid.of(txId);
+                assertThat(actualVgtid).isNotEqualTo(baseVgtid);
+            }
+            assertRecordEnd(expectedTxId, expectedRecordsCount);
```

Comment:
Can we add another assertion to ensure the vgtid legitimately changes on the commit boundary? A regular (non-batch) transaction would be fine.